mantidproject/mantid
# More Cppcheck use stl alg warning fixes

### Description of work

Dealing with the remaining (there are ~4 left which I think aren't worth changing) cppcheck warnings for using stl algorithms (first part #37205).

Should help make code safer and faster, as well as chipping away at the cppcheck suppressions backlog.
This was also a bit of training for me to build confidence with the stl algorithms.


*There is no associated issue.*


### To test:

- Tests should pass
- Once again, careful code review needed. Obviously in an ideal world any behaviour changes would result in tests failing, but we know that's not always the case.
- Cppcheck is now warning that the function `makeMappings` from `SaveNexusProcessed.cpp` is now only going to return false (specifically the call at line 572) I can't see how this is the case, so I've suppressed the `knownConditionTrueFalse` warning. I would appreciate someone else checking though.




---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
